PUBSTWO-1719 (Include Compiler contributors in Crossref XML (as an ed…

orosz-usgs opened the pull request:
…itor))

  Updated CrossrefTransformer and supporting code to include Complier contributors in the cross reference Xml
  Updated tests
  Clean up double ;

Commit:
PUBSTWO-1719 (Include Compiler contributors in Crossref XML (as an editor))
  Updated CrossrefTransformer and supporting code to include Complier contributors in the cross reference Xml
  Updated tests
  Clean up double ;

dsteinich commented on src/test/java/gov/usgs/cida/pubs/busservice/CrossRefBusServiceTest.java:
```diff
@@ -246,16 +247,35 @@ public void tetBuildCrossRefPost() throws IOException {
 
 	@Test
 	public void getGoodCrossrefXml() throws XMLValidationException, IOException {
-		when(contributorTypeDao.getById(ContributorType.AUTHORS)).thenReturn(buildContributorTypeAuthor());
```
Can the PublicationBusService.class & ContributorType.class be removed from the @SpringBootTest and contributorTypeDao also be removed?

orosz-usgs replied:
yup

dsteinich commented on src/main/java/gov/usgs/cida/pubs/domain/ContributorType.java:
```diff
@@ -17,6 +17,7 @@
 	public static final Integer COMPILERS = 3;
 	public static String AUTHOR_KEY = "authors";
 	public static String EDITOR_KEY = "editors";
+	public static String COMPILER_KEY = "compilers";
```
A block should be added in onApplicationEvent to get this value from the database.
Not the biggest fan of this pattern, but the text can be changed in the database and we need to actual value from there to tie everything together. These values up here are more just a default convenience for testing...

Commit:
PUBSTWO-1719 (Include Compiler contributors in Crossref XML (as an editor))
  Removed unneeded classes from test CrossRefBusServiceTest
  Added setting of COMPILER_KEY to onApplicationEvent
